forte/train_preprocessor.py

```diff
@@ -0,0 +1,355 @@
+#  Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+from typing import Optional, Dict, Type, Any, Union, Iterator, List
+from texar.torch.data import DataIterator, Batch
+import torch
+from torch import device
+
+from forte.common.configuration import Config
+from forte.data.converter import Converter
+from forte.data.data_pack import DataPack
+from forte.data.data_pack_dataset import DataPackDataSource, \
+    DataPackDataset
+from forte.data.extractor.base_extractor import BaseExtractor
+from forte.data.ontology.core import Entry
+from forte.data.ontology.core import EntryType
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "TrainPreprocessor"
+]
+
+
+class TrainPreprocessor:
+    r"""
+    `TrainPreprocessor` provides the functionality of doing pre-processing work
```

Comment:
The important part of this documentation is to allow users to understand how to use this class, i.e. how to pass in extractors, requests, and how they work. One way to do it is to provide a simple example here.

Reply:
Since now we added one more layer `Trainer` that will internally make use of `TrainPreprocessor` and `Trainer` is the class users should directly use, I think we should add this documentation in the `Trainer`?

Reply:
I currently add this part in the [`BaseTrainer` class](https://github.com/asyml/forte/pull/297/commits/674ed004ffc595ec3f013b112aeed2bef53308f5). If you don't think this is suitable, please let me know.